src/components/change-wallet/AddressRow.tsx

```diff
@@ -0,0 +1,336 @@
+  Edit: 'edit',
+  Notifications: 'notifications',
+  Remove: 'remove',
+};
+
+interface AddressRowProps {
+  contextMenuActions: EditWalletContextMenuActions;
+  data: any;
+  editMode: boolean;
+  onPress: () => void;
+}
+
+export default function AddressRow({
+  contextMenuActions,
+  data,
+  editMode,
+  onPress,
+}: AddressRowProps) {
+  const notificationsEnabled = useExperimentalFlag(NOTIFICATIONS);
+
+  const {
+    address,
+    balance,
+    color: accountColor,
+    ens,
+    image: accountImage,
+    isSelected,
+    isReadOnly,
+    label,
+    walletId,
+  } = data;
+
+  const { colors, isDarkMode } = useTheme();
+
+  let cleanedUpBalance = balance;
+  if (balance === '0.00') {
+    cleanedUpBalance = '0';
+  }
+
+  const cleanedUpLabel = useMemo(() => removeFirstEmojiFromString(label), [
+    label,
+  ]);
+
+  const emoji = useMemo(
+    () =>
+      returnStringFirstEmoji(cleanedUpLabel) ||
+      profileUtils.addressHashedEmoji(address),
+    [address, cleanedUpLabel]
+  );
+
+  const displayAddress = useMemo(
+    () => abbreviations.address(toChecksumAddress(address) || '', 4, 6),
+    [address]
+  );
+
+  const walletName = cleanedUpLabel || ens || displayAddress;
+
+  const linearGradientProps = useMemo(
+    () => ({
+      ...gradientProps,
+      colors: [
+        colors.alpha(colors.blueGreyDark, 0.03),
+        colors.alpha(colors.blueGreyDark, isDarkMode ? 0.02 : 0.06),
+      ],
+      end: { x: 1, y: 1 },
+      start: { x: 0, y: 0 },
+    }),
+    [colors, isDarkMode]
+  );
+
+  const menuConfig = useMemo(() => {
+    const menuItems = notificationsEnabled
+      ? [
+          {
+            actionKey: ContextMenuKeys.Remove,
+            actionTitle: lang.t('wallet.action.remove'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'trash.fill',
+            },
+            menuAttributes: ['destructive'],
+          },
+          {
+            actionKey: ContextMenuKeys.Notifications,
+            actionTitle: lang.t('wallet.action.notifications'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'bell.fill',
+            },
+          },
+          {
+            actionKey: ContextMenuKeys.Edit,
+            actionTitle: lang.t('wallet.action.edit'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'pencil',
+            },
+          },
+        ]
+      : [
+          {
+            actionKey: ContextMenuKeys.Remove,
+            actionTitle: lang.t('wallet.action.remove'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'trash.fill',
+            },
+            menuAttributes: ['destructive'],
+          },
+          {
+            actionKey: ContextMenuKeys.Edit,
+            actionTitle: lang.t('wallet.action.edit'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'pencil',
+            },
+          },
+        ];
```

Comment:
menu items could be stored as constants outside of the component and based on platform already properly ordered, like in reverse or not. The reverse will happen just once so it should be allright to so

Reply:
then in the component without using useMemo  we could pick the right variant for notifications enabled or notifications disabled